### ISSUE TITLE
Update/Set parameters(filters,visuals)

### ISSUE THREAD

**Issue by benito2313** [pbi-tools/pbi-tools#104]
Hi,

I would like to change/create parameters with pbi tools.
Changing/creating the following:
Filters
Visuals
Pages
Bookmarks
Selections

My challenge now is that i have set a filter value on the report filterpane and i would like to change it to something else.
And also turn the visuals filter visibility to false would be awesome.

Is there a way to do this with pbi-tools.

Kind regards,

Ben

**Comment by mthierba** [comment]
I want this in `pbi-tools` as much as you do, just juggling various commitments.

I know for sure that what you're describing is possible since I'm already running those exact same transforms outside of `pbi-tools`. Let's get there in little steps: What would be the bare minimum to support your scenario? Change a report or page level filter?

**Comment by benito2313** [comment]
Yes the bare minimum would be to change the report/page/visual filter .
In the mean while I created a powershell script that does this for me.
And it also sets the lock on the filters and changes if it’s visible or not.

**Comment by BVMather** [comment]
> 

Agree Tarek this is often an issue. I think of parameters as a way to generate multiple versions of a report from a single master. My thinking for the start page was more generally how to set up unit tests that could check certain report properties, the start page being one of them.

**Comment by TarekSalha** [comment]
Sure, I just wanted to give a very basic example 😀
BTW, I am doing this exact unit test currently using tabular editor best practice analyzer as workaround.

**Comment by BVMather** [comment]
Wow, is that possible to analyse the actual report definition using BPA?  I thought you could only query the model itself. I'd love to know more!

**Comment by TarekSalha** [comment]
Hi Brian,
I have to apologize. I talked to my teammates and they confirmed, that we are indeed not able to access this property via BPA as it is not part of the report. Sorry, my bad!
Actually, in the past, we did the following: The report section of pbi-tools `extract` outcome is basically contained 1:1 inside the pbix. You can immediately see this by changing the file extension of any report from pbix to zip and unpack it. We were reading the json into memory and performing unit tests about it. (expected number of visuals, number of pages, active startpage, etc...)
Currently, we are switching this to pbi-tools as it is much more convenient than parsing the original files from the zip.
